link_resolver_ffz.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"text/template"
+	"time"
+)
+
+var (
+	errInvalidFFZEmotePath = errors.New("invalid FrankerFaceZ emote path")
+)
+
+func init() {
+	const (
+		emoteAPIURL = "https://api.frankerfacez.com/v1/emote/%s"
+
+		thumbnailFormat = "https://cdn.frankerfacez.com/emoticon/%s/4"
+
+		tooltipTemplate = `<div style="text-align: left;">
+<b>{{.Code}}</b><br>
+<b>FrankerFaceZ Emote</b><br>
+<b>By:</b> {{.Uploader}}</div>`
+	)
+
+	var (
+		emoteNotFoundResponse = &LinkResolverResponse{
+			Status:  http.StatusNotFound,
+			Message: "No FrankerFaceZ emote with this id found",
+		}
+	)
+
+	type TooltipData struct {
+		Code     string
+		Uploader string
+	}
+
+	type FrankerFaceZEmoteAPIResponse struct {
+		Height       int16  `json:"height"`
+		Modifier     bool   `json:"modifier"`
+		Status       int    `json:"status"`
+		Width        int16  `json:"width"`
+		Hidden       bool   `json:"hidden"`
+		CreatedAtRaw string `json:"created_at"`
+		CreatedAt    time.Time
+		UpdatedAtRaw string `json:"last_updated"`
+		UpdatedAt    time.Time
+		ID           int    `json:"id"`
+		Name         string `json:"name"`
+		Public       bool   `json:"public"`
```

Comment:
I don't wanna be picky, but do we need to specify unused things like `Width`, `Height` or `Modifier` in the struct? I do realize we might make use of some properties in the future if / when we would want to modify this tooltip in the future, but stuff like `Modifier` seem unnecessary to me.

Reply:
Generally Go structs are autogenerated (like I believe this is) - I think it's fine for us when doing that to just keep everything even if we don't necessarily need it. Keeps updating/diffing responses easier in the future